Use pkg-config to detect libxml2.

libxml2 >= 2.12 no longer installs the aclocal macro which defines
AM_PATH_XML2, so the autotools bootstrap in eautoreconf died with:

    You must have libxml2 (>= 2.6.0) installed
    autom4te: error: /usr/bin/m4 failed with exit status: 1

Adapted from the Debian patch
debian/patches/0004-autotools-use-pkg-config-to-test-libxml2.patch
(https://sources.debian.org/patches/dvdauthor/0.7.2-3/) by Hugh McMaster;
the XML2_ variable naming matches upstream master
(https://github.com/ldo/dvdauthor/blob/master/configure.ac).
Bug: https://github.com/macaroni-os/mark-issues/issues/764

diff -ruN orig/configure.ac mod/configure.ac
--- a/configure.ac	2016-12-30 23:25:38.000000000 -0000
+++ b/configure.ac	2026-10-25 19:56:02.214494658 -0000
@@ -135,10 +135,11 @@
 
 AM_ICONV
 
-ifdef([AM_PATH_XML2],
-    [AM_PATH_XML2(2.6.0, , AC_MSG_ERROR([You must have libxml2 >= 2.6.0 installed]))],
-    [errprint([You must have libxml2 (>= 2.6.0) installed
-])m4_exit(1)AC_MSG_ERROR([You must have libxml2 (>= 2.6.0) installed])])
+dnl Replaces the AM_PATH_XML2 check. libxml2 >= 2.12 stopped installing the
+dnl aclocal macro that defines AM_PATH_XML2, which broke the autotools
+dnl bootstrap (macaroni-os/mark-issues#764). Use pkg-config detection,
+dnl mirroring upstream master (ldo/dvdauthor) and Debian 0.7.2-3.
+PKG_CHECK_MODULES([XML2], [libxml-2.0 >= 2.6.0])
 
 AC_CHECK_DECLS(O_BINARY, , , [ #include <fcntl.h> ] )
 
diff -ruN orig/src/Makefile.am mod/src/Makefile.am
--- a/src/Makefile.am	2016-10-25 00:26:59.000000000 -0000
+++ b/src/Makefile.am	2026-10-25 19:56:02.218300965 -0000
@@ -8,7 +8,7 @@
 nodist_bin_SCRIPTS = dvddirdel
 
 AM_CPPFLAGS = -DSYSCONFDIR="\"$(sysconfdir)\"" \
-    @XML_CPPFLAGS@ @MAGICK_CPPFLAGS@ @FREETYPE_CPPFLAGS@ @FRIBIDI_CFLAGS@ @FONTCONFIG_CFLAGS@
+    @XML2_CFLAGS@ @MAGICK_CPPFLAGS@ @FREETYPE_CPPFLAGS@ @FRIBIDI_CFLAGS@ @FONTCONFIG_CFLAGS@
 AM_CFLAGS = -Wall
 
 dvdauthor_SOURCES = dvdauthor.c common.h dvdauthor.h da-internal.h \
@@ -16,17 +16,17 @@
     dvdifo.c dvdvob.c dvdpgc.c \
     dvdcli.c readxml.c readxml.h \
     conffile.c conffile.h compat.c compat.h rgb.h
-dvdauthor_LDADD = $(LIBICONV) $(XML_LIBS)
+dvdauthor_LDADD = $(LIBICONV) $(XML2_LIBS)
 
 dvdunauthor_SOURCES = dvdunauthor.c dvduncompile.c common.h dvduncompile.h compat.c compat.h
-dvdunauthor_LDADD = $(XML_LIBS) $(LIBICONV) -ldvdread
+dvdunauthor_LDADD = $(XML2_LIBS) $(LIBICONV) -ldvdread
 
 spumux_SOURCES = subgen.c subgen.h rgb.h \
     subgen-parse-xml.c readxml.c readxml.h \
     subgen-encode.c subgen-image.c subglobals.h \
     conffile.c conffile.h compat.c compat.h common.h \
     subrender.c subrender.h subreader.c subreader.h subfont.c subfont.h
-spumux_LDADD = $(XML_LIBS) $(MAGICK_LIBS) $(FREETYPE_LIBS) \
+spumux_LDADD = $(XML2_LIBS) $(MAGICK_LIBS) $(FREETYPE_LIBS) \
     $(FRIBIDI_LIBS) $(FONTCONFIG_LIBS) $(LIBICONV) -lm
 
 spuunmux_SOURCES = spuunmux.c rgb.h compat.c compat.h common.h conffile.h conffile.c
